### PR TITLE
feat: display sdgs tiles in a 2 column grid

### PR DESCRIPTION
**Pull request by Iansong95**
## Purpose

Display sdgs page in a 2-column grid format

## Changes

1 big change, 1 small change
Big change: I switched from ScrollView to FlatList (couldn't get ScrollView to format for the life of me)
Small change: because FlatList uses "item" as a variable for each entry of data, I had to change the almost everything related to the sdg array, i.e. sdg.number -> item.number
This also affected the individual sdg pages.

### Screenshots

image

## Checklist

- [x] Tested on Android
- [ ] Tested on iOS

### CODE REVIEW

File: screens/sdgs/index.js [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-            <TileText adjustsFontSizeToFit numberOfLines={1} allowFontScaling>
-              {sdg.name}
-            </TileText>
-          </TileContainer>
-        ))}
-      </ScrollView>
+      <FlatList
+        style={{ width: '100%' }}
+        data={sdgs}
+        renderItem={({ item }) => <Item item={item} />}
```
Comment (BenjaminYong) [inline screens/sdgs/index.js:31]: `<Item item={item} />`
can be changed to
`<Item sdg={item} />`

File: screens/sdgs/index.js [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
-const TileText = styled(Text)`
-  color: white;
-  font-size: 25px;
-  margin: 5px;
-`;
-
 function SdgsScreen({ navigation }) {
+  const numColumns = 2;
+
+  const Item = ({ item }) => (
```
Comment (BenjaminYong) [inline screens/sdgs/index.js:19]: `const Item = ({ item }) => (`
can be changed to
`const Item = ({ sdg }) => (`

File: screens/sdgs/index.js [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
+    <TileContainer onPress={() => navigation.navigate('SDG', { item: item })}>
```
Comment (BenjaminYong) [inline screens/sdgs/index.js:20]: Any `item` in here can be changed to `sdg`- this is your custom item component so you can define what 'item' is.

File: screens/sdgs/index.js [hunk 4]
```diff
@@ -1,37 +1,37 @@
...
+      <Image source={item.src} resizeMode="contain" style={{ height: '100%', width: '100%', borderRadius: 20 }} />
```
Comment (BenjaminYong) [inline screens/sdgs/index.js:21]: Same goes with `item here`

File: screens/sdgs/index.js [hunk 5]
```diff
@@ -1,37 +1,37 @@
...
-              {sdg.name}
-            </TileText>
-          </TileContainer>
-        ))}
-      </ScrollView>
+      <FlatList
+        style={{ width: '100%' }}
+        data={sdgs}
+        renderItem={({ item }) => <Item item={item} />}
+        keyExtractor={(item) => item.number}
```
Comment (BenjaminYong) [inline screens/sdgs/index.js:32]: This cannot be changed to sdg as the `keyExtractor `property is expecting item - all good

File: screens/sdg/index.js [hunk 6]
```diff
@@ -4,11 +4,11 @@ import { Text } from 'react-native';
 import ChildScreen from '../../components/child-screen';
 
 const Sdg = ({ route }) => {
-  const { sdg } = route.params;
+  const { item } = route.params;
```
Comment (BenjaminYong) [inline screens/sdg/index.js:7]: Once you change the names in  the sdgs/index.js file, you can rename all `item` in here back to `sdg`

**Review by BenjaminYong — CHANGES_REQUESTED** [review]
Nice stuff Ian, good find on the FlatList react component.
You can still keep the use of 'sdg' as shown in my comments.
'item' is still required in the FlatList properties parameters, but only required there.

**Comment by jonowu** [comment]
I removed border-radius because it didn't look right on ios for some reason. We'll definitely investigate why and add it back after the demo today.

**Review by jonowu — APPROVED** [review]

**Pull request merged**